hedera-node/hedera-app/src/xtest/java/contract/AssortedOpsXTest.java

```diff
@@ -0,0 +1,288 @@
+    }
+
+    private TransactionBody synthCreateTxn() {
+        return TransactionBody.newBuilder()
+                .transactionID(TransactionID.newBuilder().accountID(MISC_PAYER_ID))
+                .contractCreateInstance(ContractCreateTransactionBody.newBuilder()
+                        .autoRenewPeriod(STANDARD_AUTO_RENEW_PERIOD)
+                        .fileID(ASSORTED_OPS_INITCODE_FILE_ID)
+                        .gas(GAS_TO_OFFER)
+                        .build())
+                .build();
+    }
+
+    private TransactionBody synthLazyCreateTxn() {
+        return TransactionBody.newBuilder()
+                .transactionID(TransactionID.newBuilder().accountID(RELAYER_ID))
+                .ethereumTransaction(EthereumTransactionBody.newBuilder()
+                        .ethereumData(ETH_LAZY_CREATE)
+                        .maxGasAllowance(10 * ONE_HBAR)
+                        .build())
+                .build();
+    }
+
+    private TransactionBody synthDeterministicDeploy() {
+        final var encoded = DEPLOY_DETERMINISTIC_CHILD.encodeCallWithArgs(SALT);
+        return TransactionBody.newBuilder()
+                .transactionID(TransactionID.newBuilder().accountID(MISC_PAYER_ID))
+                .contractCall(callWithParams(ASSORTED_OPS_CONTRACT_ID, encoded))
+                .build();
+    }
+
+    private TransactionBody synthGoldbergesqueDeploy() {
+        return callTransaction(ASSORTED_OPS_CONTRACT_ID, DEPLOY_GOLDBERGESQUE.encodeCallWithArgs(SALT));
+    }
+
+    private TransactionBody synthVacateAddress() {
+        return callTransaction(FINALIZED_AND_DESTRUCTED_CONTRACT_ID, VACATE_ADDRESS.encodeCallWithArgs());
+    }
+
+    private TransactionBody synthTakeFive() {
+        return callTransaction(ASSORTED_OPS_CONTRACT_ID, TAKE_FIVE.encodeCallWithArgs());
+    }
+
+    private TransactionBody callTransaction(@NonNull final ContractID contractId, @NonNull final ByteBuffer encoded) {
+        return TransactionBody.newBuilder()
+                .transactionID(TransactionID.newBuilder().accountID(MISC_PAYER_ID))
+                .contractCall(callWithParams(contractId, encoded))
+                .build();
+    }
+
+    private ContractCallTransactionBody callWithParams(
+            @NonNull final ContractID contractId, @NonNull final ByteBuffer encoded) {
+        return ContractCallTransactionBody.newBuilder()
+                .functionParameters(Bytes.wrap(encoded.array()))
+                .contractID(contractId)
+                .gas(GAS_TO_OFFER)
+                .build();
+    }
+
+    @Override
+    protected long initialEntityNum() {
+        return NEXT_ENTITY_NUM - 1;
+    }
+
+    @Override
+    protected Map<FileID, File> initialFiles() {
+        final var files = new HashMap<FileID, File>();
+        files.put(
+                ASSORTED_OPS_INITCODE_FILE_ID,
+                File.newBuilder()
+                        .contents(resourceAsBytes("initcode/AssortedXTest.bin"))
```

Comment:
Do we really need 3 copies of `AssortedXTest.bin`.  Can't this one refer to the one in the bytecode directory?  Or does this one include the initcode?

Reply:
Removed the _test-clients/_ assets other than `.sol` since they aren't being used in any `HapiSpec`'s in CI.

> Or does this one include the initcode?

Right, this is the initcode.